### PR TITLE
[Improve] Beautify the YOLOv7 configuration

### PR DESCRIPTION
**Pull request by tianleiSHI**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
不是很确定数据增强部分的参数注释是否合适
Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py [hunk 1]
```diff
@@ -1,42 +1,103 @@
...
-num_classes = 80
+num_det_layers = 3  # The number of model output scales
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
+
+# Data augmentation
+max_translate_ratio = 0.2  # YOLOv5RandomAffine
+scaling_ratio_range = (0.1, 2.0)  # YOLOv5RandomAffine
+
+mixup_alpha = 8.0,  # YOLOv5MixUp
+mixup_beta = 8.0  # YOLOv5MixUp
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:77]: randchoice_mosaic_pipeline 里面的 0.8 0.2 也要写到这，这个应该是会修改的

Comment (hhaAndroid) [reply configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:77]: train_cfg 中的dynamic_intervals 也要放到频繁修改参数里面去，270 这个数放到上面，这个参数表示最后 30 epoch 开始评估间隔为1了。建议 270 -> max_epoch-30 比较好

File: configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py [hunk 2]
```diff
@@ -68,7 +76,7 @@
         type='YOLOv5MixUp',
         alpha=8.0,
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py:77]: l 配置里面既然把这两个参数提到前面，那么这里也要提的，保持一致

Comment (hhaAndroid) [reply configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py:77]: 还有 lr_factor 参数也要

File: configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py [hunk 3]
```diff
@@ -1,10 +1,14 @@
 _base_ = './yolov7_l_syncbn_fast_8x16b-300e_coco.py'
 
+# ========================modified parameters========================
+# -----data related-----
 img_scale = (1280, 1280)  # height, width
 num_classes = 80
 # only on Val
 batch_shapes_cfg = dict(img_size=img_scale[0], size_divisor=64)
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py:8]: 注释

Comment (tianleiSHI) [reply configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py]: 已增加注释

File: configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py (hunk 2 above)
Comment (tianleiSHI) [reply configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py]: ok已修改

File: configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py (hunk 1 above)
Comment (tianleiSHI) [reply configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:82]: 已修改

File: configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py [hunk 4]
```diff
@@ -1,42 +1,108 @@
...
 
-# different from yolov5
-anchors = [
-    [(12, 16), (19, 36), (40, 28)],  # P3/8
-    [(36, 75), (76, 55), (72, 146)],  # P4/16
-    [(142, 110), (192, 243), (459, 401)]  # P5/32
-]
+# -----model related-----
+# The scaling factor that controls the depth of the network structure
+deepen_factor = 0.33
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:68]: 没有这个参数

File: configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py [hunk 5]
```diff
@@ -1,42 +1,108 @@
...
+# The scaling factor that controls the width of the network structure
+widen_factor = 0.5
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:70]: 没有这个参数

File: configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py [hunk 6]
```diff
@@ -1,19 +1,42 @@
...
+strides = [8, 16, 32, 64]  # Strides of multi-scale prior box
+num_det_layers = 4  # # The number of model output scales
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
 
+# Data augmentation
+max_translate_ratio = 0.2,  # YOLOv5RandomAffine
+scaling_ratio_range = (0.1, 2.0),  # YOLOv5RandomAffine
+mixup_prob = 0.15  # YOLOv5MixUp
+randchoice_mosaic_prob = [0.8, 0.2]
+mixup_alpha = 8.0,  # YOLOv5MixUp
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py:31]:
```suggestion
mixup_alpha = 8.0  # YOLOv5MixUp
```

File: configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py [hunk 7]
```diff
@@ -1,19 +1,42 @@
...
+mixup_beta = 8.0,  # YOLOv5MixUp
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py:32]:
```suggestion
mixup_beta = 8.0  # YOLOv5MixUp
```

File: configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py [hunk 8]
```diff
@@ -1,19 +1,42 @@
...
 ]
-strides = [8, 16, 32, 64]
-num_det_layers = 4
+strides = [8, 16, 32, 64]  # Strides of multi-scale prior box
+num_det_layers = 4  # # The number of model output scales
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
 
+# Data augmentation
+max_translate_ratio = 0.2,  # YOLOv5RandomAffine
+scaling_ratio_range = (0.1, 2.0),  # YOLOv5RandomAffine
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py:28]:
```suggestion
scaling_ratio_range = (0.1, 2.0)  # YOLOv5RandomAffine
```

File: configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py [hunk 9]
```diff
@@ -1,19 +1,42 @@
...
     [(436, 615), (739, 380), (925, 792)]  # P6/64
 ]
-strides = [8, 16, 32, 64]
-num_det_layers = 4
+strides = [8, 16, 32, 64]  # Strides of multi-scale prior box
+num_det_layers = 4  # # The number of model output scales
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
 
+# Data augmentation
+max_translate_ratio = 0.2,  # YOLOv5RandomAffine
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_w-p6_syncbn_fast_8x16b-300e_coco.py:27]:
```suggestion
max_translate_ratio = 0.2  # YOLOv5RandomAffine
```

File: configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py [hunk 10]
```diff
@@ -1,10 +1,26 @@
...
 
+# ========================modified parameters========================
+
+# -----model related-----
+# Data augmentation
+max_translate_ratio = 0.1  # YOLOv5RandomAffine
+scaling_ratio_range = (0.5, 1.6)  # YOLOv5RandomAffine
+mixup_prob = 0.05  # YOLOv5MixUp
+randchoice_mosaic_prob = [0.8, 0.2]
+mixup_alpha = 8.0,  # YOLOv5MixUp
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_tiny_syncbn_fast_8x16b-300e_coco.py:11]:
```suggestion
mixup_alpha = 8.0  # YOLOv5MixUp
```

File: configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py [hunk 11]
```diff
@@ -1,42 +1,108 @@
...
+num_det_layers = 3  # The number of model output scales
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
 
-# single-scale training is recommended to
+# Data augmentation
+max_translate_ratio = 0.2  # YOLOv5RandomAffine
+scaling_ratio_range = (0.1, 2.0)  # YOLOv5RandomAffine
+mixup_prob = 0.15  # YOLOv5MixUp
+randchoice_mosaic_prob = [0.8, 0.2]
+mixup_alpha = 8.0,  # YOLOv5MixUp
```
Comment (hhaAndroid) [inline configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py:81]:
```suggestion
mixup_alpha = 8.0  # YOLOv5MixUp
```

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**